### ISSUE TITLE
Idea

### ISSUE THREAD

**Issue by azhe825** [hil-se/pairwise-ranking#1]
## Context
Training data: pre-fix code and post-fix code pairs.

## Approach
 - Train an encoder that outputs one value f(x) = (0-1).
 - We know that f(pre) > f(post) and f(pre) is vulnerable. But f(post) can also be vulnerable.
 - Loss function: 
   + max{margin - (f(pre) - f(post)), 0}
   + max{thres - f(pre), 0}
